python/ray/job_config.py

```diff
@@ -39,6 +39,7 @@ def __init__(self,
         self.metadata = metadata or {}
         self.ray_namespace = ray_namespace
         self.set_runtime_env(runtime_env)
+        self._default_actor_lifetime = None
```

Comment:
Missing parameter in `__init__`. And the default value should be `non-detached`.